### PR TITLE
Commiting the initial code for PR

### PR DESCRIPTION
**Pull request by samrdd**
Commited the initial code, sql file is good, python file done upto statistics. Final model is not done yet. Kindly review. Will commit the final code in a while.

### CODE REVIEW

File: Asignment7/baseball_stats.sql [hunk 1]
```diff
@@ -0,0 +1,485 @@
+-- Using the baseball database
+USE baseball;
+
+-- Show all the tables in the baseball database
+SHOW TABLES;
+
+-- appending the local date to the team_batting_stats and team_pitching_stats and storing in a table
+
+DROP TABLE IF EXISTS s_team_batting_stats;
```
Comment (panchalashish4) [inline Asignment7/baseball_stats.sql:9]: I will prefer using temporary tables instead of creating tables in DB for creating features so you don't create unnecessary tables which you won't be using later. Only the final table needs to be in DB.

File: Asignment7/baseball_stats.sql [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+               team_pitching_counts.plateApperance as p_plateApperance
+               FROM team_pitching_counts
+        JOIN game
+        ON game.game_id = team_pitching_counts.game_id;
+
+-- calculating the rolling batting average of each team of 200 days
+
+DROP TABLE IF EXISTS s_team_bat_rolling_avg;
+CREATE TABLE s_team_bat_rolling_avg
+    (INDEX Game_id_IND(game_id),INDEX Team_id_IND(team_id), INDEX t_rolling_batting_average(t_rolling_battingaverage))
```
Comment (panchalashish4) [inline Asignment7/baseball_stats.sql:51]: You can use ENGINE=MEMORY before SELECT to calculate the rolling average even more faster.

File: Asignment7/baseball_stats.sql [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+           t1.game_id,
+           t1.local_date,
+           IF((sum(t2.b_AtBat) - SUM(t2.b_StrikeOut) - SUM(t2.b_Home_Run) + SUM(t2.b_Sac_Fly)) = 0, 0,
+                    (SUM(t2.b_Hit) - SUM(t2.b_Home_Run))/(sum(t2.b_AtBat) - SUM(t2.b_StrikeOut) - SUM(t2.b_Home_Run) + SUM(t2.b_Sac_Fly))) as b_team_BABIP
+        FROM s_team_batting_stats AS t1 JOIN s_team_batting_stats AS t2
+        ON t1.team_id= t2.team_id
+        WHERE t2.local_date >= DATE_ADD(t1.local_date, INTERVAL -200 DAY)
+        AND t2.local_date< t1.local_date
+     GROUP BY t1.team_id, t1.local_date,t1.game_id;
+SELECT * FROM team_rollingBABIP LIMIT 0,20;
```
Comment (panchalashish4) [inline Asignment7/baseball_stats.sql:108]: line 108 and 109: You can remove these statements as you don't have that table.

File: Asignment7/baseball_stats.sql [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+               t1.local_date,
+               IF((sum(t2.p_AtBat) - SUM(t2.p_StrikeOut) - SUM(t2.p_Home_Run) + SUM(t2.p_Sac_Fly)) = 0, 0,
+                            (SUM(t2.p_Hit) - SUM(t2.p_Home_Run))/(sum(t2.p_AtBat) - SUM(t2.p_StrikeOut) - SUM(t2.p_Home_Run) + SUM(t2.p_Sac_Fly))) as p_team_BAIP
+            FROM s_team_pitching_stats AS t1 JOIN s_team_pitching_stats AS t2
+            ON t1.team_id= t2.team_id
+            WHERE t2.local_date >= DATE_ADD(t1.local_date, INTERVAL -200 DAY)
+            AND t2.local_date< t1.local_date
+         GROUP BY t1.team_id, t1.local_date,t1.game_id;
+
+-- Calculating the rolling BB/k(walk to strike out ratio for batter) for teams
```
Comment (panchalashish4) [inline Asignment7/baseball_stats.sql:127]: I can see you have created multiple tables for creating individual features for the batting and pitching teams. You can calculate all features in one single temp table together for the batting and pitching teams respectively.

File: Asignment7/baseball_stats.sql [hunk 5]
```diff
@@ -0,0 +1,485 @@
...
+           s_home_team_HCP.home_team_HCP as home_team_HCP,
+           s_starting_pitcher_rolling_WHIP.starting_pitcher_rolling_WHIP as home_team_startingpitcher_WHIP,
+           s_starting_pitcher_K_per_9inn.k_per_9_innings as home_team_startingpitcher_K_per_9_inn,
+           s_starting_pitcher_BB_per_9inn.BB_per_9_innings as home_team_startingpitcher_BB_per_9_inn,
+           s_starting_pitcher_H_per_9inn.H_per_9_innings as home_team_startingpitcher_H_per_9_inn,
+           s_baseball_game_stats.winner
+    FROM s_baseball_game_stats
+    JOIN s_team_bat_rolling_avg
+    ON s_baseball_game_stats.home_team_id = s_team_bat_rolling_avg.team_id and s_baseball_game_stats.game_id = s_team_bat_rolling_avg.game_id
+    JOIN s_team_home_runs_scored
```
Comment (panchalashish4) [inline Asignment7/baseball_stats.sql:339]: You can make code more readable by splitting AND statements in multiple lines.

File: Asignment7/baseball_stats.sql [hunk 6]
```diff
@@ -0,0 +1,485 @@
...
+               IF (SUM(t2.innings_pitched)=0,0,(SUM(t2.Hit) + SUM(t2.Walk))/(SUM(t2.innings_pitched))) AS starting_pitcher_rolling_WHIP
+            FROM s_starting_pitcher_table AS t1 JOIN s_starting_pitcher_table AS t2
+            ON t1.team_id= t2.team_id
+            WHERE t2.local_date >= DATE_ADD(t1.local_date, INTERVAL -200 DAY)
+            AND t2.local_date< t1.local_date
+         GROUP BY t1.team_id, t1.local_date,t1.game_id;
+
+-- appending game and boxscore table
+
+DROP TABLE IF EXISTS s_baseball_game_stats;
```
Comment (panchalashish4) [inline Asignment7/baseball_stats.sql:258]: If you're taking a response from winner_home_or_away from the boxscore table, you might want to cross-check the results as I realized that values in that column are not always correct. Take a look:
`SELECT game_id, home_runs, away_runs, winner_home_or_away
FROM boxscore
WHERE game_id IN (333417, 325875)
ORDER BY game_id;`

File: Asignment7/baseball_stats.sql [hunk 7]
```diff
@@ -0,0 +1,485 @@
...
+
+-- Updating the winner column by replacing H and A with 1 and 0
+
+UPDATE baseball_game_stats
+SET winner = REPLACE(winner,'A','0');
+UPDATE baseball_game_stats
+SET winner = REPLACE(winner,'H','1');
+
+
+SELECT * FROM baseball_game_stats LIMIT 0,20;
```
Comment (panchalashish4) [inline Asignment7/baseball_stats.sql:466]: You should take a look into your JOINS as the final table only has game_ids which have values and dropped other game ids. Take a look:
`SELECT * FROM baseball_game_stats
ORDER BY game_id;`

File: Asignment7/Assignment7.py [hunk 8]
```diff
@@ -10,7 +10,9 @@
 from scipy.stats import chi2_contingency
```
Comment (panchalashish4) [inline Asignment7/Assignment7.py]: pre-commit hooks failed for the code.

File: Asignment7/Assignment7.py [hunk 9]
```diff
@@ -1057,6 +1010,77 @@ def main():
...
+
+    sql_engine = sqlalchemy.create_engine(connect_string)
+
+    query = """
+                    SELECT * FROM baseball_game_stats
+                    """
+
+    df = pandas.read_sql_query(query, sql_engine)
+
+    predictors = ['HT_BA',
```
Comment (panchalashish4) [inline Asignment7/Assignment7.py:171]: I'm unable to run your python code as the predictors' name doesn't match your final SQL table. I think you forgot to commit your updated SQL code.

File: Asignment7/Assignment7.py [hunk 10]
```diff
@@ -1057,6 +1010,77 @@ def main():
...
+
+    get_all_tables(df, predictors, response, Name)
+
+    """Splitting the data to test and train and applying a decision tree model """
+
+    X = df[predictors]
+
+    y=df[response]
+
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.33, random_state = 42)
```
Comment (panchalashish4) [inline Asignment7/Assignment7.py]: You might want to take a look at your train_test_split and add:
`shuffle=False`

File: Asignment7/Assignment7.py [hunk 11]
```diff
@@ -1057,6 +1010,77 @@ def main():
...
+
+    model = DecisionTreeClassifier(criterion='gini', max_depth=3, random_state=0)
```
Comment (panchalashish4) [inline Asignment7/Assignment7.py]: Only one model is implemented, use multiple models and compare their performances.

File: Asignment7/Assignment7.py [hunk 12]
```diff
@@ -1057,6 +1010,77 @@ def main():
...
+
+    # fit the model
+    model.fit(X_train, y_train)
+
+    y_pred = model.predict(X_test)
+
+    print('Model accuracy score with criterion gini index: {0:0.4f}'.format(accuracy_score(y_test, y_pred)))
 
 if __name__ == "__main__":
```
Comment (panchalashish4) [inline Asignment7/Assignment7.py:269]: I'm unable to run the code but the functionalities from hw4 and midterm look good.

**Review by panchalashish4 — CHANGES_REQUESTED** [review]
Take a look at the comments and make modifications as required.
